group-coordinator/src/test/java/org/apache/kafka/coordinator/group/OffsetMetadataManagerTest.java

```diff
@@ -1963,6 +2080,75 @@ public void testFetchAllOffsetsAtDifferentCommittedOffset() {
         ), context.fetchAllOffsets("group", Long.MAX_VALUE));
     }
 
+    @Test
+    public void testFetchAllOffsetsWithPendingTransactionalOffsets() {
+        OffsetMetadataManagerTestContext context = new OffsetMetadataManagerTestContext.Builder().build();
+
+        context.groupMetadataManager.getOrMaybeCreateConsumerGroup("group", true);
+
+        context.commitOffset("group", "foo", 0, 100L, 1);
+        context.commitOffset("group", "foo", 1, 110L, 1);
+        context.commitOffset("group", "bar", 0, 200L, 1);
+
+        context.commit();
+
+        assertEquals(3, context.lastWrittenOffset);
+        assertEquals(3, context.lastCommittedOffset);
+
+        context.commitOffset(10L, "group", "foo", 1, 111L, 1, context.time.milliseconds());
+        context.commitOffset(10L, "group", "bar", 0, 201L, 1, context.time.milliseconds());
```

Comment:
should we have bar-1 again to test that case too?

Reply:
Oh interesting. Is it worth testing that we don't return it for now and update it if we plan to change the behavior?

Reply:
I could add it but it does not bring much value in my opinion.

Reply:
I would think it is just codifying the state of the protocol and will flag us if it changes.

Reply:
That’s fair. Let me add it tomorrow.